heltour/tournament/views.py

```diff
@@ -615,62 +614,61 @@ def view(self, username):
         pairings = player.pairings.exclude(scheduled_time=None)
         return self.ical_from_pairings_list(pairings, calendar_title, uid_component)
 
-class RegisterView(LeagueView):
+class RegisterView(LoginRequiredMixin, LeagueView):
+
     def view(self, post=False):
-        if self.season is not None and self.season.registration_open:
-            reg_season = self.season
-        else:
-            reg_season = Season.objects.filter(league=self.league, registration_open=True).order_by('-start_date').first()
+        reg_season = Season.get_registration_season(self.league, self.season)
         if reg_season is None:
             return self.render('tournament/registration_closed.html', {})
-        if post:
-            form = RegistrationForm(self.request.POST, season=reg_season)
-            if form.is_valid():
-                with reversion.create_revision():
-                    reversion.set_comment('Submitted registration.')
-                    form.save()
-                # send registration received email
-                subject = render_to_string('tournament/emails/registration_received_subject.txt', {'reg': form.instance})
-                msg_plain = render_to_string('tournament/emails/registration_received.txt', {'reg': form.instance})
-                msg_html = render_to_string('tournament/emails/registration_received.html', {'reg': form.instance})
-                try:
-                    send_mail(
-                        subject,
-                        msg_plain,
-                        settings.DEFAULT_FROM_EMAIL,
-                        [form.cleaned_data['email']],
-                        html_message=msg_html,
-                    )
-                except SMTPException:
-                    logger.exception('A confirmation email could not be sent.')
-
-                self.request.session['reg_email'] = form.cleaned_data['email']
-                return redirect(leagueurl('registration_success', league_tag=self.league.tag, season_tag=self.season.tag))
-        else:
-            form = RegistrationForm(season=reg_season)
-            if self.request.user.is_authenticated():
-                player = Player.get_or_create(self.request.user.username)
-                form.fields['lichess_username'].initial = player.lichess_username
-                form.fields['email'].initial = player.email
-                form.fields['classical_rating'].initial = player.rating_for(reg_season.league)
-                form.fields['has_played_20_games'].initial = not player.provisional_for(reg_season.league)
-                form.fields['already_in_slack_group'].initial = player.slack_user_id != ''
 
-        context = {
-            'form': form,
-            'registration_season': reg_season
-        }
-        return self.render('tournament/register.html', context)
+        with cache.lock(f'update_create_registration-{self.request.user.id}-{reg_season.id}'):
```

Comment:
The registration page needs some kind of protection if the user's last registration was rejected. For now maybe just redirect to the league home page.